File: src/com/tsunderebug/discordintellij/ToggleAction.scala
package com.tsunderebug.discordintellij

import javax.swing.Icon

import com.intellij.openapi.actionSystem.{AnAction, AnActionEvent}
import com.intellij.openapi.util.IconLoader

class ToggleAction extends AnAction {

  override def actionPerformed(a: AnActionEvent): Unit = {
    val pe = a.getProject.getComponent[PresenceEnabled](classOf[PresenceEnabled], new PresenceEnabled())
    pe match {
      case PresenceEnabled(true) =>
        hideRPC()
        pe.loadState(PresenceEnabled(false))
        a.getPresentation.setIcon(ToggleAction.disabled)
      case PresenceEnabled(false) =>
        pe.loadState(new PresenceEnabled())
        a.getPresentation.setIcon(ToggleAction.enabled)
        enableRPC(a.getProject)
    }
  }

}

object ToggleAction {

  val enabled: Icon = IconLoader.findIcon("/icons/enabled.png")
  val disabled: Icon = IconLoader.findIcon("/icons/disabled.png")

}

File: src/com/tsunderebug/discordintellij/ToggleAction.java
package com.tsunderebug.discordintellij;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.util.IconLoader;

import javax.swing.Icon;

public class ToggleAction extends AnAction {

	public static final Icon ENABLED = IconLoader.findIcon("/icons/enabled.png");
	public static final Icon DISABLED = IconLoader.findIcon("/icons/disabled.png");

	@Override
	public void actionPerformed(AnActionEvent a) {
		PresenceEnabled pe = a.getProject().getComponent(PresenceEnabled.class, new PresenceEnabled());
		if(pe.isEnabled()) {
			pe.loadState(new PresenceEnabled(false));
			a.getPresentation().setIcon(ToggleAction.DISABLED);
			DiscordIntelliJ.hideRPC();
		} else {
			pe.loadState(new PresenceEnabled());
			a.getPresentation().setIcon(ToggleAction.ENABLED);
			DiscordIntelliJ.enableRPC(a.getProject());
		}
	}

}
